ethereum/statetransition/src/test/java/tech/pegasys/artemis/statetransition/AttestationAggregatorTest.java

```diff
@@ -0,0 +1,166 @@
+  }
+
+  @Test
+  void addOwnValidatorAttestation_oldData_noNewAttester() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    BLSSignature sig = attestation.getAggregate_signature();
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation newAttestation = new Attestation(attestation);
+    newAttestation.setAggregate_signature(BLSSignature.random());
+    aggregator.addOwnValidatorAttestation(newAttestation);
+    assertEquals(aggregator.getAggregateAndProofs().size(), 1);
+    assertEquals(
+        sig, aggregator.getAggregateAndProofs().get(0).getAggregate().getAggregate_signature());
+  }
+
+  @Test
+  void addOwnValidatorAttestation_oldData_newAttester() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    BLSSignature sig1 = attestation.getAggregate_signature();
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation newAttestation = AttestationGenerator.withNewSingleAttesterBit(attestation);
+    BLSSignature sig2 = BLSSignature.random();
+    newAttestation.setAggregate_signature(sig2);
+    aggregator.addOwnValidatorAttestation(newAttestation);
+    assertEquals(aggregator.getAggregateAndProofs().size(), 1);
+    assertEquals(
+        aggregator.getAggregateAndProofs().get(0).getAggregate().getAggregate_signature(),
+        BLSAggregate.bls_aggregate_signatures(List.of(sig1, sig2)));
+  }
+
+  @Test
+  void processAttestation_newData_noOwnValidatorAttestationExists() throws Exception {
+    Attestation attestation = attestationGenerator.validAttestation(storageClient);
+    BLSSignature sig = attestation.getAggregate_signature();
+    int validatorIndex = new Random().nextInt(1000);
+    aggregator.committeeIndexToAggregatorInformation.put(
+        attestation.getData().getIndex(),
+        new AggregatorInformation(BLSSignature.random(), validatorIndex));
+    aggregator.addOwnValidatorAttestation(attestation);
+    Attestation newAttestation = new Attestation(attestation);
+    newAttestation.setData(
+        new AttestationData(
+            attestation.getData().getSlot().plus(UnsignedLong.ONE), attestation.getData()));
+    newAttestation.setAggregate_signature(BLSSignature.random());
+    aggregator.processAttestation(newAttestation);
+    assertEquals(aggregator.getAggregateAndProofs().size(), 1);
+    assertEquals(
+        sig, aggregator.getAggregateAndProofs().get(0).getAggregate().getAggregate_signature());
```

Comment:
Can we just check that the attestation is equal to the original here?
```suggestion
        attestation, aggregator.getAggregateAndProofs().get(0).getAggregate());
```

Reply:
great idea.

Reply:
done.